dbt_platform_helper/providers/config.py

```diff
@@ -0,0 +1,105 @@
+import os
+from copy import deepcopy
+from pathlib import Path
+
+import click
+import yaml
+from schema import SchemaError
+from yaml.parser import ParserError
+from yamllint import linter
+from yamllint.config import YamlLintConfig
+
+from dbt_platform_helper.constants import PLATFORM_CONFIG_FILE
+from dbt_platform_helper.providers.platform_config_schema import PlatformConfigSchema
+from dbt_platform_helper.utils.messages import abort_with_error
+
+
+class ConfigProvider:
+    def __init__(self, config_validator, config=None, echo=click.secho):
+        self.config = config or {}
+        self.validator = config_validator
+        self.echo = echo
+
+    # TODO - this is to do with Yaml validation
+    def lint_yaml_for_duplicate_keys(self, file_path: str, lint_config=None):
```

Comment:
this function is returning a list of duplicate keys.  I would like to rename it.  I think we will be moving it into a YamlFileProvider class anyway so probably don't need to touch it just now.